### DISCUSSION TITLE
How to bind to "Super" key only?

### DISCUSSION THREAD

**Discussion by mkkl-dev** [Ulauncher/Ulauncher#922] [Q&A] [answered]
Hi! 

Awesome launcher and features. Really love it!

However, it seems I can't bind the **Super** key to launch it? I can bind "Super + 'something'" but not Super-only. Is there anyway this can be achieved?

**Using Ubuntu 18.10 by the way**

**Comment by friday** [comment]
Super is a modifier. It doesn't trigger a normal key press event. If you want it to act like a plain key when you press and release immediately without hitting any other keys this is called "dual-role" keys. You have to use something like `xcape` (X11 only) for that. ~~There's no good alternative for wayland~~.

Update 2022: There is now actually. A tool called Interception tools and the [Dual Function Keys](https://gitlab.com/interception/linux/plugins/dual-function-keys) plugin.

**Comment by aasmpro** [comment]
using `debian buster` with `xfce4` DE, installed latest version of ulauncher `v4.3.1.r4` and set Hotkey to `Super L`, it works great.

Edit:
by the way, it does not allow any other `Super L + <key>` shortcuts.

**Comment by gornostal** [comment]
@aasmpro thanks for providing a workaround.

> Can't bind to "Super" key only
That's s by design as explained by @friday.

I'm closing this issue.

**Comment by friday** [comment, marked as answer]
I just learned you can do this in Gnome too if you unset Mutter's `overlay-key` and then bind to `Super_L`. You have to use dconf or gsettings since the GUI keyboard settings is restricted by the same reason as already mentioned.

This should work:

```
CKBPATH=org.gnome.settings-daemon.plugins.media-keys.custom-keybinding:/org/gnome/settings-daemon/plugins/media-keys/custom-keybindings
# Disable default mapping for Super 
gsettings set org.gnome.mutter overlay-key ''
# Add custom keybinding for Ulauncher
gsettings set $CKBPATH/ulauncher/ name 'ulauncher'
gsettings set $CKBPATH/ulauncher/ command 'ulauncher-toggle'
gsettings set $CKBPATH/ulauncher/ binding 'Super_L'
```

While it doesn't break Super L as a modifier key it will show ulauncher unconditionally when you press Super, not like Mutter's `overlay-key`, which is implemented as a real dual-role key. I suspect some or hard coded logic is to blame for the inconsistency. `xcape` is still my recommendation.

Restoring the defaults:

```
CKBPATH=org.gnome.settings-daemon.plugins.media-keys.custom-keybinding:/org/gnome/settings-daemon/plugins/media-keys/custom-keybindings
gsettings reset org.gnome.mutter overlay-key
gsettings reset-recursively $CKBPATH/ulauncher/

```

**Reply by luispablo** [reply]
This solution, despite working, broke all my other Super+&lt;key&gt; bindings.

**Reply by friday** [reply]
@luispablo. That's a shame. It's nothing we can fix in Ulauncher of course. Maybe try https://github.com/rvaiya/keyd or https://github.com/kmonad/kmonad (I haven't done so myself, I'm using https://gitlab.com/interception/linux/plugins/dual-function-keys, but I think these newer options are easier).

**Reply by Blackstar1069** [reply]
Hello, i've found a workaround by installing an extension called "Super Key" [https://extensions.gnome.org/extension/5188/super-key/](url)

... no problems so far.

**Comment by Vondelphia** [comment]
Hey @friday - might you know how to use xcape to get this to work, specifically? I was [browsing over the github](https://github.com/alols/xcape) but the usage part is like gibberish. Maybe I have to slow down :smirk: :joy: 

I run / install in terminal: `sudo dnf copr enable dawid/xcape` ... and than install xcape and run `xcape -e 'Super_L=F1'` but when I press SUPER key it doesn't fire off an F1 (which is what I have bound to launch Ulauncher).

Thanks in advance and hope all is well.

**Reply by friday** [reply]
@Vondelphia I hope you found a solution. Xcape is designed to give users a lot of flexibility and power, as the cost of being very technical to adopt and learn. I recommend it if people truly want this behavior, but I'm not someone who can help you set it up.

**Comment by aindriu80** [comment]
mapping the super key to launch Ulauncher would be great

**Comment by Vaults** [comment]
For Linux Mint; a workaround is to go into `Keyboard -> Shortcuts` and add a custom shortcut to `ulauncher-toggle` with `Super-L`. Works as a keypress and doesn't obfuscate other shortcuts.

**Comment by skjerns** [comment]
I found out how to set the `Super`/`Meta` key in KDE Plasma 5.24.7 to toggle Ulauncher while retaining the ability to use other shortcuts such as Super+D or Super+E :)

the trick is to assign the dbus call to the `~/.config/kwinrc` under the section `ModifierOnlyShortcuts` (which is available from Plasma 5.8 onwards)

however this alone will not focus the application. for this we need to install `wmctrl` and add  a script to listen to the dbus call. You can do this by these commands:

```bash
sudo apt install wmctrl -y

echo $'#!/bin/bash

dbus-monitor --profile "interface='net.launchpad.ulauncher',member='toggle_window_focus'" |
while read -r line; do
    ulauncher-toggle
done
' > ~/.config/autostart/listen_ulauncher_toggle.sh

# create autostart launcher for ulauncher toggle
echo $'[Desktop Entry]
Exec=/home/simon/.config/autostart/listen_ulauncher_toggle.sh
Icon=dialog-scripts
Name=listen_ulauncher_toggle.sh
Path=
Type=Application
X-KDE-AutostartScript=true
' > ~/.config/autostart/listen_ulauncher_toggle.sh.desktop

# register dbus to toggle and focus window
kwriteconfig5 --file ~/.config/kwinrc --group ModifierOnlyShortcuts --key Meta "net.launchpad.ulauncher,/net/launchpad/ulauncher,net.launchpad.ulauncher,toggle_window_focus"
qdbus org.kde.KWin /KWin reconfigure

```

let me know if something seems wrong, I'm quite new to Linux

**Comment by joaoseixas88** [comment]
using fedora with gnome, the most easy way that I found was install that https://extensions.gnome.org/extension/5188/super-key/ extension, then follow this https://github.com/Ulauncher/Ulauncher/wiki/Hotkey-In-Wayland tutorial and set my super key to ulauncher-toggle
